carma_wm/include/carma_wm/CARMAWorldModel.h

```diff
@@ -107,6 +114,32 @@ class CARMAWorldModel : public WorldModel
    */
   void setRouteEndPoint(const lanelet::BasicPoint3d& end_point);
 
+  /**
+   * \brief (non-const version) Gets the underlying lanelet, given the cartesian point on the map 
+   *
+   * \param point         Cartesian point to check the corressponding lanelet
+   * \param n             Number of lanelets to return. Default is 10. As there could be many lanelets overlapping.
+   * \throw std::invalid_argument if the map is not set, contains no lanelets
+   *
+   * \return vector of underlying lanelet, empty vector if it is not part of any lanelet
+   */
+  std::vector<lanelet::Lanelet> getLaneletsFromPoint(const lanelet::BasicPoint2d& point, const unsigned int n);
+
+  /**
+   * \brief (non-const version) Given the cartesian point on the map, tries to get the opposite direction lanelet on the left
+   *        This function is intended to find "adjacentLeft lanelets" that doesn't share points between lanelets
+   *        where adjacentLeft of lanelet library fails
+   *
+   * \param point         Cartesian point to check the corressponding lanelet
+   * \param n             Number of lanelets to return. Default is 10. As there could be many lanelets overlapping.
+   * 
+   * \throw std::invalid_argument if the map is not set, contains no lanelets
+   * NOTE:  Only to be used on 2 lane, opposite direction road. Number of points in all linestrings are assumed to be roughly the same.
```

Comment:
Does this method break if this condition is not met? Is there a way for us to detect that and throw and exception?

Reply:
- If the number of points in all linestrings are drastically different, it might give wrong lanelets depending on how different the numbers are.
- adding an exception is a good idea.

Reply:
I still need to add exception here